core/src/main/java/io/questdb/cairo/AbstractIntervalDataFrameCursor.java

```diff
@@ -63,8 +63,15 @@ protected static long search(ReadOnlyColumn column, long value, long low, long h
                 low = mid + 1;
             else if (midVal > value)
                 high = mid;
-            else
+            else {
+                // In case of multiple equal values, find the first
+                int increment = scanUp ? -1 : 1;
+                for (long offset = 0; (mid + offset) > 0 && (mid + offset) < high; offset += increment) {
+                    if (midVal != column.getLong((mid + offset) * 8))
+                        return mid + offset - increment;
```

Comment:
logic is great, but
```
mid + offset
```
is calculated at least 3 times per loop. I hope Java would realise that and execute only once, but I would do this kind of optimisation manually

Reply:
Fixed.